async_service/trio.py

```diff
@@ -15,14 +15,13 @@
 class TrioManager(BaseManager):
     # A nursery for sub tasks and services.  This nursery is cancelled if the
     # service is cancelled but allowed to exit normally if the service exits.
-    _task_nursery: trio_typing.Nursery
+    _task_nursery: trio_typing.Nursery = None
```

Comment:
mypy complains about this because None is not a Nursery. Do you know how I can work around that, @pipermerriam ?

Reply:
Maybe leave it unset here and change the other code block to either do a `hasattr` check **or** do a `try/except AttributeError`.  I think I slightly prefer the `hasattr` but either would be fine.

Reply:
Couldn't you type it as `Optiona[trio_typing.Nursery]` instead?

Reply:
No, because then it complains when I try to access any attributes of _task_nursery:

```
async_service/trio.py:137: error: Item "None" of "Optional[Nursery]" has no attribute "cancel_scope"
async_service/trio.py:189: error: Item "None" of "Optional[Nursery]" has no attribute "start_soon"
```

Reply:
> Couldn't you type it as `Optiona[trio_typing.Nursery]` instead?

I think that would mean you would have to appease `mypy` everywhere you did something like `self._task_nursery.something` since it would not know whether it was `None` or a nursery without some type guarding or something.

Reply:
Yeah, in this case you would need to have a single getter that would do the `is not None` check and return a non-optional `Nursery` then.